components/emoji_picker/emoji_picker.test.js

```diff
@@ -0,0 +1,406 @@
+        await wrapper.vm.$nextTick();
+
+        expect(skinToneSelectorImg.exists()).toBe(true);
+        expect(skinToneSelectorImg.attributes('aria-label')).toBe(':wave_tone2:');
+      });
+
+      it('Should display skin tone prop if it changes', async () => {
+        expect(skinToneSelectorImg.exists()).toBe(true);
+        expect(skinToneSelectorImg.attributes('aria-label')).toBe(':wave_tone1:');
+
+        await skinToneSelectorButton.trigger('click');
+        await wrapper.vm.$nextTick();
+        await skinToneMediumLightButton.trigger('click');
+        await wrapper.vm.$nextTick();
+
+        expect(skinToneSelectorImg.exists()).toBe(true);
+        expect(skinToneSelectorImg.attributes('aria-label')).toBe(':wave_tone2:');
+
+        await wrapper.setProps({ skinTone: 'Dark' });
+
+        expect(skinToneSelectorImg.exists()).toBe(true);
+        expect(skinToneSelectorImg.attributes('aria-label')).toBe(':wave_tone5:');
+      });
+    });
+
+    describe('When recently emojis is not provided', () => {
+      it('Should not render recently emojis tabset', () => {
+        mockProps = {
+          ...baseProps,
+          recentlyUsedEmojis: [],
+          tabSetLabels: [
+            'Smileys and people',
+            'Nature',
+            'Food',
+            'Activity',
+            'Travel',
+            'Objects',
+            'Symbols',
+            'Flags',
+          ],
+        };
+
+        updateWrapper();
+
+        const firstButton = wrapper.find('.d-tablist').findAll('button').at(0);
+
+        expect(firstButton.exists()).toBe(true);
+        expect(firstButton.attributes('aria-label')).not.toBe(MOCK_TAB_SET_LABELS[0]);
+        expect(firstButton.attributes('aria-label')).toBe(MOCK_TAB_SET_LABELS[1]);
+      });
+    });
+  });
+
+  describe('Interactivity Tests', () => {
+    it('Should emit selected-emoji event when emoji is clicked', async () => {
+      const emoji = wrapper.find('.d-emoji-picker__selector .d-emoji-picker__tab button');
+
+      expect(emoji.exists()).toBe(true);
+
+      await emoji.trigger('click');
+      await wrapper.vm.$nextTick();
+
+      expect(wrapper.emitted('selected-emoji')).toBeTruthy();
+    });
+
+    describe('On search', () => {
+      it('Fixed label should change on search input', async () => {
+        const searchInput = wrapper.find('.d-emoji-picker__search input');
+
+        searchInput.setValue('tube');
+
+        await searchInput.trigger('input');
+        await wrapper.vm.$nextTick();
+
+        const fixedLabel = wrapper.find('.d-emoji-picker__search-label');
+
+        expect(fixedLabel.text()).toBe(MOCK_SEARCH_RESULTS_LABEL);
+      });
+
+      it('Input search should filter emojis', async () => {
+        const searchInput = wrapper.find('.d-emoji-picker__search input');
+
+        expect(wrapper.find('[data-qa="filtered-emojis"]').exists()).toBe(false);
+
+        searchInput.setValue('tube');
+
+        await wrapper.vm.$nextTick();
```

Comment:
if you await setValue you shouldn't have to do nextTick. or does that still not work?

Reply:
Ye, definetly all `.set` or `.trigger` should we go with an await.

Will refactor it a little bit.